packages/alfa-rules/test/sia-r8/rule.spec.tsx

```diff
@@ -197,3 +197,159 @@ test("evaluate() is inapplicable for an element which is not displayed", async (
+    }),
+  ]);
+});
+
+test("evaluate() passes an input element with type=password and aria-labelledby", async (t) => {
+  const target = <input type="password" aria-labelledby="country" />;
+
+  const label = <div id="country">Country</div>;
+
+  const document = h.document([label, target]);
+
+  t.deepEqual(await evaluate(R8, { document }), [
+    passed(R8, target, {
+      1: Outcomes.InputElementWithNoAriaRoleHasName("password"),
+    }),
+  ]);
+});
+
+test("evaluate() passes an input element with type=password and placeholder attribute", async (t) => {
+  const target = <input type="password" placeholder="Your search query" />;
+
+  const document = h.document([target]);
+
+  t.deepEqual(await evaluate(R8, { document }), [
+    passed(R8, target, {
+      1: Outcomes.InputElementWithNoAriaRoleHasName("password"),
+    }),
+  ]);
+});
+
+test("evaluate() fails an input element with type=password and empty aria-label", async (t) => {
+  const target = <input type="password" aria-label=" " />;
+
+  const document = h.document([target]);
+
+  t.deepEqual(await evaluate(R8, { document }), [
+    failed(R8, target, {
+      1: Outcomes.InputElementWithNoAriaRoleHasNoName("password"),
+    }),
+  ]);
+});
+
+test(`evaluate() fails an input element with type=password and aria-labelledby pointing to an
+     empty element`, async (t) => {
+  const target = <input type="password" aria-labelledby="country" />;
+
+  const label = <div id="country"></div>;
+
+  const document = h.document([label, target]);
+
+  t.deepEqual(await evaluate(R8, { document }), [
+    failed(R8, target, {
+      1: Outcomes.InputElementWithNoAriaRoleHasNoName("password"),
+    }),
+  ]);
+});
+
+test(`evaluate() is inapplicable for an input element with type=password 
+    and aria-hidden`, async (t) => {
+  const target = <input type="password" aria-hidden="true" aria-label="firstname" />;
+
+  const document = h.document([target]);
+
+  t.deepEqual(await evaluate(R8, { document }), [inapplicable(R8)]);
+});
+
+test(`evaluate() is inapplicable for an element with type=password and which 
+    is not displayed`, async (t) => {
+  const target = <input type="password" aria-label="firstname" style={{ display: "none" }} />;
+
+  const document = h.document([target]);
+
+  t.deepEqual(await evaluate(R8, { document }), [inapplicable(R8)]);
+});
+
+test(`evaluate() fails for input elements with various types which give it no ARIA 
+    role and which have no accessible name`, async (t) => {
+  const targets = [<input type="color"/>, <input type="date"/>, <input type="datetime-local"/>, 
+    <input type="file"/>, <input type="month"/>, <input type="time"/>, 
+    <input type="week"/>];
+  const document = h.document(targets)
+  t.deepEqual(await evaluate(R8, { document }), 
+    targets.map(target => failed(R8, target, { 
+      1: Outcomes.InputElementWithNoAriaRoleHasNoName(target.attribute("type")
+        .map(attr => attr.value).getOr("")) })), 
+  );
```

Comment:
**Nitpick:** I think I would rather loop through the attributes to show the similarity in tests.
```suggestion
  for (const type of ["color", "date", "datetime-local", "file", "month", "time", "week"]) {
    const target = <input type="{type}"/>;
    const document = h.document([target])
    t.deepEqual(await evaluate(R8, { document }), 
      failed(R8, target, { 
        1: Outcomes.InputElementWithNoAriaRoleHasNoName(target.attribute("type")
          .map(attr => attr.value).getOr("")) }), 
      );
    }
```
(the types list may even be extracted as a global variable in this file to be shared between both tests)

Reply:
Done in cedb1925cf7db8d0c14af553e206376e94841712